core/src/main/clojure/xtdb/indexer.clj

```diff
@@ -525,8 +527,8 @@
 
                   ^:volatile-mutable indexer-error
 
-                  ^PriorityBlockingQueue awaiters]
-
+                  ^PriorityBlockingQueue awaiters
+                  ^PersistentHashMap metrics]
```

Comment:
use `^Map` here - if the map's small enough, you're not guaranteed to get a PHM in Clojure.

(if you even need the type-hint at all, that is)